[lametrictime] Error using JAX-RS client from CXF

Using lametrictime on openHAB 3, which uses CXF instead of Jersey, I get the following error in my log:
```
14:24:56.042 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - Thing 'lametrictime:device:47ee543589' changed from INITIALIZING to UNKNOWN
14:24:56.365 [ERROR] [org.apache.cxf.jaxrs.utils.JAXRSUtils] - Problem with reading the data, class java.lang.String, ContentType: application/json;charset=UTF8.
14:24:56.366 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - Thing 'lametrictime:device:47ee543589' changed from UNKNOWN to OFFLINE (COMMUNICATION_ERROR): Unable to connect to LaMetric Time
```
@wborn fyi

--- a/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/common/impl/AbstractClient.java
+++ b/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/common/impl/AbstractClient.java
@@ -20,31 +20,24 @@ import javax.ws.rs.client.ClientBuilder;
 
 import com.google.gson.Gson;
 
-public abstract class AbstractClient
-{
-    private final ClientBuilder clientBuilder;
+public abstract class AbstractClient {
+    protected final ClientBuilder clientBuilder;
 
     private volatile Client client;
     private volatile Gson gson;
 
-    public AbstractClient()
-    {
+    public AbstractClient() {
         this(ClientBuilder.newBuilder());
     }
 
-    public AbstractClient(ClientBuilder clientBuilder)
-    {
+    public AbstractClient(ClientBuilder clientBuilder) {
         this.clientBuilder = clientBuilder;
     }
 
-    protected Client getClient()
-    {
-        if (client == null)
-        {
-            synchronized (this)
-            {
-                if (client == null)
-                {
+    protected Client getClient() {
+        if (client == null) {
+            synchronized (this) {
+                if (client == null) {
                     client = createClient();
                 }
             }
@@ -53,14 +46,10 @@ public abstract class AbstractClient
         return client;
     }
 
-    protected Gson getGson()
-    {
-        if (gson == null)
-        {
-            synchronized (this)
-            {
-                if (gson == null)
-                {
+    protected Gson getGson() {
+        if (gson == null) {
+            synchronized (this) {
+                if (gson == null) {
                     gson = createGson();
                 }
             }
@@ -71,13 +60,11 @@ public abstract class AbstractClient
 
     protected abstract Client createClient();
 
-    protected Gson createGson()
-    {
+    protected Gson createGson() {
         return GsonGenerator.create();
     }
 
-    protected ClientBuilder getClientBuilder()
-    {
+    protected ClientBuilder getClientBuilder() {
         return clientBuilder;
     }
 }

--- a/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/local/impl/LaMetricTimeLocalImpl.java
+++ b/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/local/impl/LaMetricTimeLocalImpl.java
@@ -18,6 +18,8 @@ package org.openhab.binding.lametrictime.api.local.impl;
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
 import java.security.KeyManagementException;
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
@@ -25,7 +27,6 @@ import java.security.cert.X509Certificate;
 import java.util.List;
 import java.util.SortedMap;
 import java.util.logging.Logger;
-import java.util.stream.Collectors;
 
 import javax.net.ssl.SSLContext;
 import javax.net.ssl.TrustManager;
@@ -131,9 +132,11 @@ public class LaMetricTimeLocalImpl extends AbstractClient implements LaMetricTim
         Response response = getClient().target(getApi().getEndpoints().getNotificationsUrl())
                 .request(MediaType.APPLICATION_JSON_TYPE).get();
 
+        Reader entity = new BufferedReader(
+                new InputStreamReader((InputStream) response.getEntity(), StandardCharsets.UTF_8));
+
         // @formatter:off
-        return getGson().fromJson(response.readEntity(String.class),
-                                  new TypeToken<List<Notification>>(){}.getType());
+        return getGson().fromJson(entity,  new TypeToken<List<Notification>>(){}.getType());
         // @formatter:on
     }
 
@@ -256,8 +259,8 @@ public class LaMetricTimeLocalImpl extends AbstractClient implements LaMetricTim
         Response response = getClient().target(getApi().getEndpoints().getAppsListUrl())
                 .request(MediaType.APPLICATION_JSON_TYPE).get();
 
-        String entity = new BufferedReader(new InputStreamReader((InputStream) response.getEntity())).lines()
-                .collect(Collectors.joining("\n"));
+        Reader entity = new BufferedReader(
+                new InputStreamReader((InputStream) response.getEntity(), StandardCharsets.UTF_8));
 
         // @formatter:off
         return getGson().fromJson(entity,

--- a/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/local/impl/LaMetricTimeLocalImpl.java
+++ b/bundles/org.openhab.binding.lametrictime/src/3rdparty/java/org/openhab/binding/lametrictime/api/local/impl/LaMetricTimeLocalImpl.java
@@ -15,6 +15,9 @@
  */
 package org.openhab.binding.lametrictime.api.local.impl;
 
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.security.KeyManagementException;
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
@@ -22,6 +25,7 @@ import java.security.cert.X509Certificate;
 import java.util.List;
 import java.util.SortedMap;
 import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
 import javax.net.ssl.SSLContext;
 import javax.net.ssl.TrustManager;
@@ -252,8 +256,11 @@ public class LaMetricTimeLocalImpl extends AbstractClient implements LaMetricTim
         Response response = getClient().target(getApi().getEndpoints().getAppsListUrl())
                 .request(MediaType.APPLICATION_JSON_TYPE).get();
 
+        String entity = new BufferedReader(new InputStreamReader((InputStream) response.getEntity())).lines()
+                .collect(Collectors.joining("\n"));
+
         // @formatter:off
-        return getGson().fromJson(response.readEntity(String.class),
+        return getGson().fromJson(entity,
                                   new TypeToken<SortedMap<String, Application>>(){}.getType());
         // @formatter:on
     }
@@ -309,7 +316,7 @@ public class LaMetricTimeLocalImpl extends AbstractClient implements LaMetricTim
 
     @Override
     protected Client createClient() {
-        ClientBuilder builder = ClientBuilder.newBuilder();
+        ClientBuilder builder = clientBuilder;
 
         // setup Gson (de)serialization
         GsonProvider<Object> gsonProvider = new GsonProvider<>();

